clicommand/util_keygen.go

```diff
@@ -0,0 +1,111 @@
+package clicommand
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/buildkite/agent/v3/internal/jwkutil"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/urfave/cli"
+	"golang.org/x/exp/slices"
+)
+
+type KeygenConfig struct {
+	Alg                  string `cli:"alg" validate:"required"`
+	KeyID                string `cli:"key-id" validate:"required"`
+	PrivateKeyFilename   string `cli:"private-key-filename" normalize:"filepath"`
+	PublicKeysetFilename string `cli:"public-keyset-filename" normalize:"filepath"`
+
+	LogLevel string `cli:"log-level"`
+	Debug    bool   `cli:"debug"`
+}
+
+var KeygenCommand = cli.Command{
+	Name:  "keygen",
+	Usage: "Generate a new key pair, used to sign and verify jobs",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:   "alg",
+			EnvVar: "BUILDKITE_AGENT_KEYGEN_ALG",
+			Usage:  fmt.Sprintf("The JWS signing algorithm to use for the key pair. Valid algorithms are: %v", ValidSigningAlgorithms),
+		},
+		cli.StringFlag{
+			Name:   "key-id",
+			EnvVar: "BUILDKITE_AGENT_KEYGEN_KEY_ID",
+			Usage:  "The ID to use for the keys generated",
+		},
+		cli.StringFlag{
+			Name:   "private-key-filename",
+			EnvVar: "BUILDKITE_AGENT_KEYGEN_PRIVATE_KEY_FILENAME",
+			Usage:  "The filename to write the private key to. Defaults to a name based on the key id in the current directory",
+		},
+		cli.StringFlag{
+			Name:   "public-keyset-filename",
+			EnvVar: "BUILDKITE_AGENT_KEYGEN_PUBLIC_KEYSET_FILENAME",
+			Usage:  "The filename to write the public keyset to. Defaults to a name based on the key id in the current directory",
+		},
```

Comment:
I think we need defaults for these. It will be a barrier to first use otherwise.